heartbeat/monitors/wrappers/summarizer/summarizer.go

```diff
@@ -84,61 +131,48 @@ func NewJobSummary(attempt uint16, maxAttempts uint16, retryGroup string) *JobSu
 	}
 }
 
+// BumpAttempt swaps the JobSummary object's pointer for a new job summary
+// that is a clone of the current one but with the Attempt field incremented.
+func (js *JobSummary) BumpAttempt() {
+	*js = *NewJobSummary(js.Attempt+1, js.MaxAttempts, js.RetryGroup)
+}
+
 // Wrap wraps the given job in such a way that the last event summarizes all previous events
 // and additionally adds some common fields like monitor.check_group to all events.
 // This adds the state and summary top level fields.
 func (s *Summarizer) Wrap(j jobs.Job) jobs.Job {
 	return func(event *beat.Event) ([]jobs.Job, error) {
-		conts, jobErr := j(event)
-
-		_, _ = event.PutValue("monitor.check_group", fmt.Sprintf("%s-%d", s.checkGroup, s.jobSummary.Attempt))
+		conts, eventErr := j(event)
 
 		s.mtx.Lock()
 		defer s.mtx.Unlock()
 
-		js := s.jobSummary
-
 		s.contsRemaining-- // we just ran one cont, discount it
 		// these many still need to be processed
 		s.contsRemaining += uint16(len(conts))
 
-		monitorStatus, err := event.GetValue("monitor.status")
-		if err == nil && !eventext.IsEventCancelled(event) { // if this event contains a status...
-			mss := monitorstate.StateStatus(monitorStatus.(string))
-
-			if mss == monitorstate.StatusUp {
-				js.Up++
-			} else {
-				js.Down++
+		for _, plugin := range s.plugins {
+			actions := plugin.EachEvent(event, eventErr)
+			if actions&DropErrEvent != 0 {
+				eventErr = nil
 			}
 		}
 
 		if s.contsRemaining == 0 {
-			if js.Down > 0 {
-				js.Status = monitorstate.StatusDown
-			} else {
-				js.Status = monitorstate.StatusUp
-			}
-
-			// Get the last status of this monitor, we use this later to
-			// determine if a retry is needed
-			lastStatus := s.stateTracker.GetCurrentStatus(s.sf)
+			var retry bool
+			for _, plugin := range s.plugins {
+				actions := plugin.OnSummary(event)
+				if actions&RetryOnSummary != 0 {
+					retry = true
+				}
 
-			// FinalAttempt is true if no retries will occur
-			js.FinalAttempt = js.Status != monitorstate.StatusDown || js.Attempt >= js.MaxAttempts
-
-			ms := s.stateTracker.RecordStatus(s.sf, js.Status, js.FinalAttempt)
-
-			eventext.MergeEventFields(event, mapstr.M{
-				"summary": js,
-				"state":   ms,
-			})
+			}
 
-			logp.L().Debugf("attempt info: %v == %v && %d < %d", js.Status, lastStatus, js.Attempt, js.MaxAttempts)
-			if !js.FinalAttempt {
-				// Reset the job summary for the next attempt
-				// We preserve `s` across attempts
-				s.jobSummary = NewJobSummary(js.Attempt+1, js.MaxAttempts, js.RetryGroup)
+			if !retry {
+				// on final run emits a metric for the service when summary events are complete
+				logger.LogRun(event)
```

Comment:
I've revamped a lot of this logic now and moved more status logic into the plugin, that said we __do__ know the status at this point since this is after all wrappers and plugins have run.

Reply:
I ll try again though, I thought the wrap common added the `monitor.status` only on summary event. Logs showed the same problem. Will retest it again.